tstest/integration: integration tests are failing on macOS

In a clean checkout, integration tests fail on my local Mac with the following error:

```console
$ ./tool/go clean -testcache; ./tool/go test ./tstest/integration/ -run=TestClientSideJailing -timeout=15s
--- FAIL: TestClientSideJailing (8.21s)
    integration.go:528: DERP httpsrv listener: 127.0.0.1:54027
    integration.go:562: control URL: http://127.0.0.1:54029
    integration.go:609: 2025/10/03 13:33:46 DEBUG-ADDR=127.0.0.1:54036
    integration_test.go:838: Running /var/folders/zt/s7fj3b9x3cj8shsdpnv3z2g80000gn/T/TestClientSideJailing1813976416/001/tailscale --socket=/var/folders/zt/s7fj3b9x3cj8shsdpnv3z2g80000gn/T/TestClientSideJailing1813976416/002/tailscale.sock up --login-server=http://127.0.0.1:54029 --reset ...
    integration.go:618: panic: runtime error: invalid memory address or nil pointer dereference
    integration.go:618: [signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x100f37270]
    integration.go:618:
    integration.go:618: goroutine 1 [running]:
    integration.go:618: github.com/tailscale/peercred.(*Creds).PID(0x1015b53e0?)
    integration.go:618: 	/Users/alexwlchan/go/pkg/mod/github.com/tailscale/peercred@v0.0.0-20250107143737-35a0c7bd7edc/peercred.go:22
    integration.go:618: tailscale.com/ipn/ipnauth.GetConnIdentity(0x14000055888?, {0x1015b53e0, 0x140002ca000})
    integration.go:618: 	/Users/alexwlchan/repos/tailscale/ipn/ipnauth/ipnauth_unix_creds.go:22 +0x150
    integration.go:618: tailscale.com/ipn/ipnserver.newActor(0x101596650, {0x1015b53e0?, 0x140002ca000?})
    integration.go:618: 	/Users/alexwlchan/repos/tailscale/ipn/ipnserver/actor.go:45 +0x28
    integration.go:618: tailscale.com/ipn/ipnserver.contextWithActor({0x1015af3b8, 0x14000332240}, 0x14000055948?, {0x1015b53e0?, 0x140002ca000?})
    integration.go:618: 	/Users/alexwlchan/repos/tailscale/ipn/ipnserver/actor.go:183 +0x38
    integration.go:618: tailscale.com/ipn/ipnserver.(*Server).Run.func4({0x1015af3b8?, 0x14000332240?}, {0x1015b53e0?, 0x140002ca000?})
    integration.go:618: 	/Users/alexwlchan/repos/tailscale/ipn/ipnserver/server.go:537 +0x40
    integration.go:618: net/http.(*Server).Serve(0x1400032e100, {0x1015ad740, 0x140002588d0})
    integration.go:618: 	/Users/alexwlchan/.cache/tsgo/aa85d1541af0921f830f053f29d91971fa5838f6/src/net/http/server.go:3485 +0x2cc
    integration.go:618: tailscale.com/ipn/ipnserver.(*Server).Run(0x1400031a300, {0x1015af3f0, 0x1400022c820}, {0x1015ad740, 0x140002588d0})
    integration.go:618: 	/Users/alexwlchan/repos/tailscale/ipn/ipnserver/server.go:541 +0x31c
    integration.go:618: main.startIPNServer({0x1015af2c0, 0x101d07b40}, 0x101596650, {0xb8, 0x1c, 0x32, 0x29, 0x1c, 0x5, 0x9c, ...}, ...)
    integration.go:618: 	/Users/alexwlchan/repos/tailscale/cmd/tailscaled/tailscaled.go:598 +0x3e0
    integration.go:618: main.run()
    integration.go:618: 	/Users/alexwlchan/repos/tailscale/cmd/tailscaled/tailscaled.go:508 +0x4c0
    integration.go:618: main.main()
    integration.go:618: 	/Users/alexwlchan/repos/tailscale/cmd/tailscaled/tailscaled.go:311 +0x9c8
    integration_test.go:838: up: failed to connect to local tailscaled (which appears to be running as tailscaled, pid 18128). Got error: Failed to connect to local Tailscale daemon for /localapi/v0/status; not running? Error: dial unix /var/folders/zt/s7fj3b9x3cj8shsdpnv3z2g80000gn/T/TestClientSideJailing1813976416/002/tailscale.sock: connect: no such file or directory
        , exit status 1
    stuntest.go:65: STUN server shutdown
FAIL
FAIL	tailscale.com/tstest/integration	8.488s
FAIL
```

I thought maybe I had some stale config or binaries lying around – I've been hacking around in `tailscale up` recently and plausibly a dev build might have had this error. No amount of cleaning fixed the error.

I did a `git bisect`, which points to c45f881 as the culprit.

I think I understand the issue, so I'm going to write a fix.

--- a/ipn/ipnauth/ipnauth_unix_creds.go
+++ b/ipn/ipnauth/ipnauth_unix_creds.go
@@ -18,12 +18,13 @@ import (
 func GetConnIdentity(_ logger.Logf, c net.Conn) (ci *ConnIdentity, err error) {
 	ci = &ConnIdentity{conn: c, notWindows: true}
 	_, ci.isUnixSock = c.(*net.UnixConn)
-	if ci.creds, err = peercred.Get(c); ci.creds != nil {
+	if creds, err := peercred.Get(c); err == nil {
+		ci.creds = creds
 		ci.pid, _ = ci.creds.PID()
 	} else if err == peercred.ErrNotImplemented {
 		// peercred.Get is not implemented on this OS (such as OpenBSD)
 		// Just leave creds as nil, as documented.
-	} else if err != nil {
+	} else {
 		return nil, err
 	}
 	return ci, nil

